docs/2-Interviews/preparing_for_interviews.md

```diff
@@ -12,22 +13,57 @@ Be prepared to:
+- Knowing how to use heap and having an idea of how they are implemented
+- Being able to implement quicksort and mergesort
+- Dynamic programming (the most advanced concept you'll probably need to know, some companies like Google love dynamic programming questions)
+- Tries, how to use them (also relatively more advanced but they show up not infrequently)
+
+If you use a textbook such as Introduction to Algorithms by Cormen, the book will often cover algorithms and data structures that are very unlikely to appear in an interview.  Things that are not that important unless you have time:
+- Any graph algorithm more advanced than Djikstra search
+- Being able to implement a balanced binary search tree from scratch
+- Fibonnaci heaps, van Emde Boas Trees, interval trees, linear programming, etc
+
+These lists can never be comprehensive, but hopefully it should provide a good starting point. Submit a GitHub issue if you want to ask about the importance of knowing a particular concept for interviews.
+
 ### Practice makes perfect!
 
 But **please please please** don't overdo it. It's so easy to feel like you can just keep preparing and studying and getting better - you can sink hours and hours and days and weeks into this, and it's not worth it. Being happy, spending time with friends, doing your assignments, etc are all very important. Definitely practice, it will help you feel more comfortable. But don't let it take over your life.
 
 If you can find friends or friendly people who are willing to interview you to help you practice, that is very valuable. Practice talking out loud and going through a problem from beginning to end.
 
-
 ### Resources:
 
-Tutorials
-
-- https://www.hackerrank.com/domains/tutorials/cracking-the-coding-interview
-- Cracking the coding interivew [pdf to old version](https://inspirit.net.in/books/placements/Cracking%20the%20Coding%20Interview.pdf) or you can buy it
+Tutorials/books
 
+- Cracking the Coding Interview ([pdf to old version](https://inspirit.net.in/books/placements/Cracking%20the%20Coding%20Interview.pdf) or you can buy it). This is the classic book. In addition to practice problems, the newer versions also cover most of the material that can show up in an interview.
+- Elements of Programming Interviews (less up-to date than CtCI but they have a few design problems if you need more advanced practice)
+- Programming Interviews Exposed
+- [Online course-version of Cracking the Coding Interview](https://www.hackerrank.com/domains/tutorials/cracking-the-coding-interview)
 
 Practice coding questions
 
-- https://www.hackerrank.com/dashboard
-- https://leetcode.com/
+- [HackerRank](https://www.hackerrank.com/dashboard) for online practice questions.
+- [LeetCode](https://leetcode.com/) for online practice questions.
+- [Pramp](https://www.pramp.com/) for the full interview experience. They match you with a peer and you take turns interviewing each other.
```

Comment:
I now also really like https://www.interviewcake.com/ - you get a few free, very well explained questions, and they email you a new one every week